src/coreclr/jit/importervectorization.cpp

```diff
@@ -147,20 +147,55 @@ GenTree* Compiler::impExpandHalfConstEqualsSIMD(
     GenTree* vec1    = gtNewIndir(simdType, gtNewOperNode(GT_ADD, TYP_BYREF, data, offset1));
     GenTree* vec2    = gtNewIndir(simdType, gtNewOperNode(GT_ADD, TYP_BYREF, gtClone(data), offset2));
 
+    GenTree* xor1;
+    GenTree* orr;
+
     if (cmpMode == OrdinalIgnoreCase)
     {
         // Apply ASCII-only ToLowerCase mask (bitwise OR 0x20 for all a-Z chars)
         GenTreeVecCon* toLowerVec1 = gtNewVconNode(simdType, toLowerMask);
         GenTreeVecCon* toLowerVec2 = gtNewVconNode(simdType, (BYTE*)toLowerMask + byteLen - simdSize);
 
-        vec1 = gtNewSimdBinOpNode(GT_OR, simdType, vec1, toLowerVec1, baseType, simdSize);
+#if defined(TARGET_XARCH)
+        if (compOpportunisticallyDependsOn(InstructionSet_AVX512F_VL))
+        {
+            GenTree* control;
+
+            control = gtNewIconNode(static_cast<uint8_t>((0xF0 | 0xCC) ^ 0xAA)); // (A | B)) ^ C
+            xor1    = gtNewSimdTernaryLogicNode(simdType, vec1, toLowerVec1, cnsVec1, control, baseType, simdSize);
```

Comment:
Shouldn't be this a sort of morph/lower pattern match optimization instead? So we can catch user patterns as well, this pattern doesn't look uncommon

Reply:
We're catching this one early because its simple/obvious. We do want to add the morph stuff as well, but that's a much more complex/involved PR. Handling these explicit cases early will also help with JIT throughput and avoid us needing to do the much more expensive general handling logic later.

If we decide just handling it in morph is better, than I can remove these explicit cases once that support is in place.